src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -1,462 +1,445 @@
+                Assert.True(Directory.Exists(Path.Combine(TestDirectory, path)));
+            });
         }
-    }
-    */
-#endif
 
-    [Fact]
-    [PlatformSpecific(PlatformID.Windows)] // whitespace in names is significant on Unix
-    public static void CreateDirectory_NonSignificantTrailingWhiteSpace_TreatsAsNonSignificant()
-    {
-        using (TemporaryDirectory directory = new TemporaryDirectory())
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)] 
+        public void WindowsNonSignificantTrailingWhiteSpace()
         {
+            // Windows will remove all nonsignificant whitespace in a path
+            DirectoryInfo testDir = Create(GetTestFilePath());
             var components = IOInputs.GetNonSignificantTrailingWhiteSpace();
 
-            foreach (string component in components)
+            Assert.All(components, (component) =>
             {
-                string path = directory.Path + component;
-
-                DirectoryInfo result = Directory.CreateDirectory(path);
+                string path = IOServices.RemoveTrailingSlash(testDir.FullName) + component;
+                DirectoryInfo result = Create(path);
 
                 Assert.True(Directory.Exists(result.FullName));
-                Assert.Equal(directory.Path, result.FullName);
-            }
+                Assert.Equal(testDir.FullName, IOServices.RemoveTrailingSlash(result.FullName));
+            });
         }
-    }
 
-    [Fact]
-    public static void CreateDirectory_ReadOnlyFileAsPath_ThrowsIOException()
-    {
-        using (TemporaryFile file = new TemporaryFile())
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        public void UnixNonSignificantTrailingWhiteSpace()
         {
-            file.IsReadOnly = true;
+            // Unix treats trailing/prename whitespace as significant and a part of the name.
+            DirectoryInfo testDir = Create(GetTestFilePath());
+            var components = IOInputs.GetNonSignificantTrailingWhiteSpace();
 
-            Assert.Throws<IOException>(() =>
+            Assert.All(components, (component) =>
             {
-                Directory.CreateDirectory(file.Path);
+                string path = IOServices.RemoveTrailingSlash(testDir.FullName) + component;
+                DirectoryInfo result = Create(path);
+
+                Assert.True(Directory.Exists(result.FullName));
+                Assert.NotEqual(testDir.FullName, IOServices.RemoveTrailingSlash(result.FullName));
             });
         }
-    }
 
-#if !TEST_WINRT // TODO: Enable once we implement WinRT file attributes
-    [Fact]
-    public static void CreateDirectory_ReadOnlyExistingDirectoryAsPath_DoesNotThrow()
-    {   // DevDivBugs #33833 
-        using (TemporaryDirectory directory = new TemporaryDirectory())
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)] // alternate data streams
+        public void PathWithAlternativeDataStreams_ThrowsNotSupportedException()
```

Comment:
Nit: alternate, not alternative (I realize this was preexisting your change and you just copied it)